pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -840,6 +844,19 @@ public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata,
     // Create field extractor
     _fieldExtractor = FieldExtractorFactory.getPlainFieldExtractor(schema);
     makeConsumerWrapper();
+
+    // Partitioners
+    _column2PartitionersMap = indexingConfig.getPartitioners();
+    if (_column2PartitionersMap != null && !_column2PartitionersMap.isEmpty()) {
+      try {
+        _nPartitions = _consumerWrapper.getPartitionCount(_kafkaTopic, /*maxWaitTimeMs=*/5000L);
```

Comment:
The realistic latency is in the order of 10's of ms. The 5s limit was in keeping with the existing code.